lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -361,27 +361,27 @@ follow nl tr cps yield header =
           where
             eT = T.pack (show e)
 
-    step :: Int -> Cursor target -> IO FollowExit
-    step delay cursor = runExceptT (nextBlocks nl cursor) >>= \case
+    step :: Int -> [BlockHeader] -> Cursor target -> IO FollowExit
```

Comment:
I think `[BlockHeader]` is a very confusing type here. Can we not have `BlockHeader` and call it say `followerTip`?

Reply:
Gets a bit weird with the `last cps0` not necessarily being apart of the chain though